node/cli/src/service.rs

```diff
@@ -110,19 +110,33 @@ construct_service_factory! {
 					local_key
 				};
 
-				executor.spawn(grandpa::run_grandpa(
-					grandpa::Config {
-						local_key,
-						// FIXME #1578 make this available through chainspec
-						gossip_duration: Duration::from_millis(333),
-						justification_period: 4096,
-						name: Some(service.config.name.clone())
+				let config = grandpa::Config {
```

Comment:
Merge 107 ~ 111 :  113~119

let config = if service.config.disable_grandpa { } else {}

Reply:
That means the rest of the config would get repeated on both branches of the `if` statement since only `local_key` differs.